python/lsst/daf/butler/core/datasets/type.py

```diff
@@ -393,6 +393,28 @@ def componentTypeName(self, component: str) -> str:
             return self.nameWithComponent(self.name, component)
         raise KeyError("Requested component ({}) not understood by this DatasetType".format(component))
 
+    def makeCompositeDatasetType(self) -> DatasetType:
+        """Return a DatasetType suitable for the composite version of this
+        component dataset type.
+
+        Returns
+        -------
+        composite : `DatasetType`
+            The composite dataset type.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if this dataset type is not a component dataset type.
+        """
+        if not self.isComponent():
+            raise RuntimeError(f"DatasetType {self.name} must be a component to form the composite")
+        name, _ = self.nameAndComponent()
+        if self.parentStorageClass is None:
+            raise ValueError(f"Parent storage class is not set. Unable to create composite type from {self}")
```

Comment:
Why `{self}` here and simply `{self.name}` above?

Reply:
Not entirely sure. Will unify.